src/gap_to_julia.jl

```diff
@@ -64,13 +64,10 @@ function gap_to_julia(t::T, x::Any) where {T<:Type}
     ## which ends up selecting the method below.
     if !(typeof(x) <: t)
         throw(ErrorException(
-            "Don't know how to convert value of type " *
-            string(typeof(x)) *
-            " to type " *
-            string(t),
+            "Don't know how to convert value of type $(typeof(x)) to type $(t)"
         ))
     end
-    return x
+    return x::T
```

Comment:
There is no `T` anymore.

Do you want this to be fast, that is, do you want it to specialize for all `T`?

Reply:
Is that you telling me very nicely that I shouldn't be an idiot and use `@nospecialize`? ;-) Because I think that sounds like a good suggestion... Thanks!